File: 2023/lua/challenge02/main.lua
local minicompiler = require("challenge02.minicompiler")

local inputFile = "challenge02/input"

local function ReadFile(filename)
  local file = io.open(filename)
  if not file then return "" end

  local contents = file:read("a")

  file:close()

  return contents
end

local fileContents = ReadFile(inputFile)

local compiler = minicompiler:new()

for i = 1, #fileContents do
  local c = fileContents:sub(i, i)
  compiler:ParseInput(c)
end

print("Solution: " .. compiler.Output)

File: 2023/cmd/challenge02/main.go
package main

import (
	"codember/internal/challenge/challenge02"
	"fmt"
	"io"
	"os"
	"strings"
)

func main() {
	fmt.Println("Running challenge02...")

	inputFile, err := os.Open("./cmd/challenge02/input")
	if err != nil {
		panic(err)
	}
	defer func(file *os.File) {
		_ = file.Close()
	}(inputFile)

	fileContents, err := io.ReadAll(inputFile)
	if err != nil {
		panic(err)
	}

	outputFile, err := os.Create("./cmd/challenge02/output")
	if err != nil {
		panic(err)
	}
	defer func(file *os.File) {
		_ = file.Close()
	}(outputFile)

	solution := challenge02.Run(strings.Trim(string(fileContents), "\n"))

	fmt.Println("Solution: ", solution)

	_, err = outputFile.WriteString(solution)
}
